server/src/main/java/org/elasticsearch/index/mapper/SourceLoader.java

```diff
@@ -112,13 +113,15 @@ class Synthetic implements SourceLoader {
          * @param fieldLoaderSupplier A supplier to create {@link SyntheticFieldLoader}, one for each leaf.
          * @param metrics Metrics for profiling.
          */
-        public Synthetic(Supplier<SyntheticFieldLoader> fieldLoaderSupplier, SourceFieldMetrics metrics) {
+        public Synthetic(Supplier<SyntheticFieldLoader> fieldLoaderSupplier, IndexSettings indexSettings, SourceFieldMetrics metrics) {
             this.syntheticFieldLoaderLeafSupplier = fieldLoaderSupplier;
             this.requiredStoredFields = syntheticFieldLoaderLeafSupplier.get()
                 .storedFieldLoaders()
                 .map(Map.Entry::getKey)
                 .collect(Collectors.toSet());
-            this.requiredStoredFields.add(IgnoredSourceFieldMapper.NAME);
+            if (indexSettings.getSkipIgnoredSourceRead() == false) {
```

Comment:
My understanding was that we only need to ignore `_ignore_source` at read time, like is done here. This to avoid the potential errors when synthesizing the source. An escape hatch. Do we need ignore `_ignore_source` at write time?

Reply:
I think it's better to add both, in case we have a bug at some point (...) with indexing. These settings won't be advertised, but we can give them to support in case we have a really urgent SDH (ingestion blocked etc).

Reply:
Ok, I see. An another escape hatch if for some reason we have a bug at index time.